src/BloomExe/ImageProcessing/RuntimeImageProcessor.cs

```diff
@@ -90,114 +89,148 @@ public string GetPathToResizedImage(string originalPath)
+		private bool GenerateThumbnail(string originalPath, string pathToProcessedImage, int newWidth)
+		{
+			try
+			{
+				using (var originalImage = PalasoImage.FromFile(originalPath))
+				{
+					// calculate dimensions
+					var newW = (originalImage.Image.Width > newWidth) ? newWidth : originalImage.Image.Width;
+					var newH = newW * originalImage.Image.Height / originalImage.Image.Width;
 
-					//if its a small image, like a creative commons logo, we don't try and resize it
-					if (originalImage.Image.Width > TargetDimension || originalImage.Image.Height > TargetDimension)
+					using (var newImg = originalImage.Image.GetThumbnailImage(newW, newH, () => false, IntPtr.Zero))
 					{
-						var maxDimension = Math.Max(originalImage.Image.Width, originalImage.Image.Height);
-						//enhance: if we had a way of knowing what the target dimension actually was, we'd use that, of course
-						shrinkFactor = (TargetDimension/(double) maxDimension);
+						newImg.Save(pathToProcessedImage);
 					}
-#endif
-					var destWidth = (int) (shrinkFactor*originalImage.Image.Width);
-					var destHeight = (int) (shrinkFactor*originalImage.Image.Height);
+				}
+			}
+			catch (OutOfMemoryException)
+			{
+				// ignore this one
+				GC.Collect();
+				return false;
+			}
 
-					using (var processedBitmap = new Bitmap(destWidth, destHeight))
-					{
-						using (Graphics g = Graphics.FromImage((Image) processedBitmap))
-						{
-#if ShrinkLargeImages
-							//in version 1.0, we used .NearestNeighbor. But if there is a border line down the right size (as is common for thumbnails that,
-							//are, for example, re-inserted into Teacher's Guides), then the line gets cut off. So I switched it to HighQualityBicubic
-							g.InterpolationMode = InterpolationMode.HighQualityBicubic; //.NearestNeighbor;//or smooth it: HighQualityBicubic
-#endif
-							var destRect = new Rectangle(0, 0, destWidth, destHeight);
-							lock (_convertWhiteToTransparent)
-							{
-								g.DrawImage(originalImage.Image, destRect, 0, 0, originalImage.Image.Width, originalImage.Image.Height,
-									GraphicsUnit.Pixel, _convertWhiteToTransparent);
-							}
-						}
+			return true;
+		}
 
-						var pathToProcessedImage = Path.Combine(_cacheFolder, Path.GetRandomFileName() + Path.GetExtension(originalPath));
 
-						//Hatton July 2012:
-						//Once or twice I saw a GDI+ error on the Save below, when the app 1st launched.
-						//I verified that if there is an IO error, that's what you get (a GDI+ error).
-						//I looked once, and the %temp%/Bloom directory wasn't there, so that's what I think caused the error.
-						//It's not clear why the temp/bloom directory isn't there... possibly it was there a moment ago
-						//but then some startup thread cleared and deleted it? (we are now running on a thread responding to the http request)
+		private bool MakeTransparentImage(string originalPath, string pathToProcessedImage)
```

Comment:
Don't like this name much. It's only the background that we (might) make transparent, and we don't even do that for jpgs. MakePngBackgroundTransparent?